src/components/Articles.jsx

```diff
@@ -1,23 +1,46 @@
-import { useContext, useEffect } from "react";
+import { useContext, useEffect, useState } from "react";
 import { ArticleContext } from "../contexts/Article";
 import { getArticles } from "../api";
 import ArticleCard from "./ArticleCard";
+import { useParams } from "react-router-dom";
 
 const Articles = () => {
   const { articles, setArticles } = useContext(ArticleContext);
+  const { article_id } = useParams();
+  const [article, setArticle] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
-    getArticles().then((articlesData) => {
-      setArticles(articlesData.articles);
-    });
-  }, []);
+    setIsLoading(true);
+    if (article_id) {
```

Comment:
Rather than having to do all this logic here where we check to see if there is an article_id, it would be better if you had a separate component for single articles. The single article is going to have votes and comments attached to it, so it will get messy if you try to keep it all in one component 👍